pep-0689.rst

```diff
@@ -0,0 +1,231 @@
+Warning about API that is changed often
+---------------------------------------
+
+The ``PyCode_New()`` family is an example of functions that are
+documented as unstable, and also often change in practice.
+
+Moving it to the semi-stable tier will make its status obvious even
+to people who don't read the docs carefully enough, and will make it
+hard to use accidentally.
+
+
+Changes during the Beta period
+------------------------------
+
+Since the API itself can change continuously up until Beta 1 (feature freeze)
+of a minor version, major users of this API are unlikely to test
+Alpha releases and provide feedback.
+It is very difficult to determine what needs to be exposed as semi-stable.
+
+Addidions to the semi-stable tier will count as *stabilization*,
+and will be allowed up to Release Candidate 1.
+
+
+Specification
+=============
+
+Several functions and types (“APIs”) will be moved to a new *semi-stable* tier.
+
+They will be expected to stay stable across patch releases,
+but may change or be removed without warning in minor releases (3.x.0),
+including Alpha and Beta releases of 3.x.0.
+
+When they change significantly, code that uses them should no longer compile
+(e.g. arguments should be added/removed, or a function should be renamed,
+but the semantic meaning of an argument should not change).
+
+Their definitions will be moved to a new directory, ``Include/semistable/``,
+and will be included from ``Python.h``.
+
+From Python 3.12 on, these APIs will only be usable when the
+``Py_USING_SEMI_STABLE_API`` macro is defined.
+CPython will only define the macro for building CPython itself
+(``Py_BUILD_CORE``).
+
+To make transition to semi-stable API easier,
+in Python 3.11 the APIs will be available without ``Py_USING_SEMI_STABLE_API``
+defined. In this case, using them will generate a deprecation warning on
+compilers that support ``Py_DEPRECATED``.
+
+A similar deprecation period will be used when making more APIs semi-stable
+in the future:
+
+- When moving from public API, the deprecation period should follow Python's
+  backwards compatibility policy (currently, it should last at least
+  two releases).
+- When moving from API that is documented as unstable,
+  the deprecation period can only last one release.
+- When moving from private API or adding new API, no deprecation period
+  is necessary.
+
+Leading underscores will be removed from the names of the moved APIs.
+The old underscored name of a renamed API will be available (as an alias
+using ``#define``) at least until that API changes.
+
+The semi-stable C-API tier and ``Py_USING_SEMI_STABLE_API`` will be documented,
+and documentation of each semi-stable API will be updated.
+
+
+Adjustments during Beta periods
+-------------------------------
+
+New APIs can be added to the semi-stable tier, and private APIs can be moved
+to it, up to the first release candidate of a new minor version.
+Consensus on the ``capi-sig`` or ``python-dev`` is needed it the beta period.
```

Comment:
```suggestion
Consensus on the ``capi-sig`` or ``python-dev`` is needed in the Beta period.
```
No strong opinion on whether to capitalize "Beta", but you capitalize it on the next line too.